.github/workflows/publish-to-pypi.yml

```diff
@@ -28,12 +23,51 @@ jobs:
       run: make install && make tests
 
     - name: Poetry bump version, build and publish
+      id: build
       shell: bash
       run: |
         proj_version=$(poetry version -s)
-        if [ $proj_version != $TAG_VERSION ]; then echo "Version $proj_version, defined in pyproject.toml, does not match TAG $TAG_VERSION of this release"; exit 3; fi
+        echo ::set-output name=PROJ_VERSION::$proj_version
         poetry update
         poetry publish --build -u __token__ -p $PYPI_TOKEN
       env:
-        TAG_VERSION: ${{ steps.context.outputs.TAG_VERSION }}
         PYPI_TOKEN: ${{ secrets.PYPI_API_TOKEN }}
+
+    - name: Check if Tag exists 🔍
```

Comment:
Instead of triggering on tag push, I just changed it to triggering on workflow_dispatch; this should allow the releaser to trigger the workflow from the UI or github cli and it will generate and apply the tags all based on the version in pyproject.toml. This should make sure that the tag, the github release and pypi release all use the same version number.

Reply:
Then if a tag already exists, the release workflow must fail hard at the beginning: version not bumped correctly and version already released.

Reply:
fixed